sdk/iot/pnp/src/az_iot_pnp_client.c

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// SPDX-License-Identifier: MIT
+
+#include <stdint.h>
+
+#include "az_iot_pnp_client.h"
+#include <az_precondition_internal.h>
+#include <az_result.h>
+#include <az_span.h>
+
+#include <_az_cfg.h>
+
+static const uint8_t pnp_client_param_separator = '&';
+static const uint8_t pnp_client_param_equals = '=';
+
+static const az_span pnp_model_id = AZ_SPAN_LITERAL_FROM_STR("digital-twin-model-id");
+
+AZ_NODISCARD az_iot_pnp_client_options az_iot_pnp_client_options_default()
+{
+  az_iot_pnp_client_options options;
+  options.user_agent = AZ_SPAN_NULL;
+  options.content_encoding = AZ_SPAN_NULL;
+  options.content_type = AZ_SPAN_NULL;
+
+  return options;
+}
+
+AZ_NODISCARD az_result az_iot_pnp_client_init(
+    az_iot_pnp_client* client,
+    az_span iot_hub_hostname,
+    az_span device_id,
+    az_span root_interface_name,
+    az_iot_pnp_client_options const* options)
+{
+  AZ_PRECONDITION_NOT_NULL(client);
+  AZ_PRECONDITION_VALID_SPAN(root_interface_name, 1, false);
```

Comment:
real real nit here: but I prefer having the preconditions in order of the function declaration (just makes it easier to eyeball and check going down the list). so in this case iot_hub_hostname -> device_id -> root-interface_name. Feel free to leave as is if that's too much

Reply:
Completely agree - fixing.